thunder/__init__.py

```diff
@@ -751,10 +816,13 @@ def fn_(*args, **kwargs) -> Any:
                 *data_for_autograd["flat_args"],
             )
             result = data_for_autograd["output"]
+            _nvtx_pop()
 
         if cache_entry.epilogue_fn:
             result, comp_to_epi = result
+            _nvtx_push("epilogue")
             cache_entry.epilogue_fn(*pro_to_epi, *comp_to_epi)
+            _nvtx_pop()
```

Comment:
A Pythonic way to do this is to add a decorator to `epilogue_fn` and it's better to do it where the function is defined instead of modifying the code here.
```py
@nvtx.annotate(domain="thunder")
def epilogue_fn():
    pass
```

Reply:
Note that epilogue_fn is generated, but last I looked
```
epilogue_fn = nvtx.annotate(domain="thunder")(epilogue_fn)
```
should do if you don't want the context manager.

Reply:
> A Pythonic way to do this is to add a decorator to `epilogue_fn` and it's better to do it where the function is defined instead of modifying the code here.
> 
> ```python
> @nvtx.annotate(domain="thunder")
> def epilogue_fn():
>     pass
> ```

There is no `epilogue_fn` in the code as this is a functor.